api/v1alpha1/condition_types.go

```diff
@@ -0,0 +1,14 @@
+package v1alpha1
+
+type ConditionType string
+
+const (
+	ReadyCondition     ConditionType = "Ready"
+	ImageResourceFound ConditionType = "ImageResourceFound"
```

Comment:
From a pool-perspective the following conditions (with state `False`, `True` and `Unknown`) make sense:

* `ImageReference`
  * `False` if the referenced image doesn't exist
  * `True` if the referenced image exist
  
  `Reason/Message`: whatever we have by hand :-)
* `PoolReady`
  * `False` if the pool exist in garm but pool-manager is false
  * `True` if pool-manager is ready
  
  `Reason/Message`: whatever we have by hand from garm-response - e.g the `LastSyncError` string?
* `ScopeReference` (in the future)
  if we want to decouple the scopereference in the poolCR (by removing the validation) - to make cr-creation idempotent